src/main/kotlin/org/rust/ide/formatter/AbstractRustFmtBlock.kt

```diff
@@ -1,26 +1,48 @@
 package org.rust.ide.formatter
 
-import com.intellij.formatting.Alignment
-import com.intellij.formatting.Block
-import com.intellij.formatting.Indent
-import com.intellij.formatting.Wrap
+import com.intellij.formatting.*
 import com.intellij.lang.ASTNode
-import com.intellij.psi.formatter.common.AbstractBlock
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.formatter.FormatterUtil
 import org.rust.lang.core.psi.RustCompositeElementTypes.MACRO_ARG
 
+/**
+ * Inspired by [com.intellij.psi.formatter.common.AbstractBlock], but rewritten for better flexibility.
+ */
 abstract class AbstractRustFmtBlock(
-    node: ASTNode,
-    alignment: Alignment?,
+    private val myNode: ASTNode,
+    private val myAlignment: Alignment?,
     private val myIndent: Indent?,
-    wrap: Wrap?,
+    private val myWrap: Wrap?,
     val ctx: RustFmtBlockContext
-) : AbstractBlock(node, wrap, alignment) {
+) : ASTBlock {
 
+    override fun getNode(): ASTNode = myNode
+    override fun getTextRange(): TextRange = node.textRange
+    override fun getAlignment(): Alignment? = myAlignment
     override fun getIndent(): Indent? = myIndent
+    override fun getWrap(): Wrap? = myWrap
+
+    override fun getSubBlocks(): List<Block> = mySubBlocks
+    private val mySubBlocks: List<Block> by lazy { buildChildren() }
```

Comment:
And yes, it is currently impossible to override Java function with Kotiln property :(